byterun/floats.c

```diff
@@ -530,6 +650,49 @@ CAMLprim value caml_log1p_float(value f)
   return caml_copy_double(caml_log1p(Double_val(f)));
 }
 
+#ifndef HAS_C99_FLOAT_OPS
+static inline double simple_erf(double x)
+{
+  double a1 =  0.254829592, a2 = -0.284496736, a3 = 1.421413741,
+         a4 = -1.453152027, a5 =  1.061405429, p  = 0.3275911,
+         t, y;
+
+  int sign = (x >= 0) ? 1 : -1;
+  x = fabs(x);
+  t = 1.0 / (1.0 + p * x);
+  y = 1.0 - (((((a5 *t  + a4) * t) + a3) * t + a2) * t + a1) * t * exp(-x * x);
```

Comment:
@jhjourdan On modern CPUs doubles will usually perform better than floats.  Unless memory size or bandwidth is an issue or if there are vectorization advantages with floats, people should use doubles all the time.

I can add a reference if we decide to go ahead with the current implementation, which is based on the formula in the book but implemented using Horner's method, which is better for evaluating polynomials.  I think John Cook is the original author of the reference code I found.

I agree that there is a small discontinuity at zero, but the jump is smaller than one billionth.  Given that we are dealing with finite representations anyway, a small degree of discontinuity is expected.

That said, of course, better approximations exist, especially around zero, where the above approximation is actually least accurate.  But then you'd have to convince the OCaml team of accepting much more complicated code.